src/main/java/coupon/coupon/business/CouponService.java

```diff
@@ -0,0 +1,33 @@
+package coupon.coupon.business;
+
+import coupon.coupon.domain.Coupon;
+import coupon.coupon.infrastructure.CouponRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Optional;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class CouponService {
+
+    private final CouponRepository couponRepository;
+    private final CouponWriterService couponWriterService;
+
+    @Transactional
+    public void create(Coupon coupon) {
+        couponRepository.save(coupon);
+    }
+
+    @Transactional(readOnly = true)
+    public Coupon getCoupon(long id) {
+        Optional<Coupon> coupon = couponRepository.findById(id);
+        if(coupon.isEmpty()) {
+            coupon = Optional.ofNullable(couponWriterService.findCouponInWriter(id));
+        }
+        return coupon.orElse(null);
```

Comment:
이렇게 개선할 수 있겠어요!
`couponWriterService.findCouponInWriter(id)`에서 null을 반환하는 경우는 없으니, optional을 nullable하게 만들어 줄 필요 없을 것 같아요~
```suggestion
        return couponRepository.findById(id)
                .orElseGet(() -> couponWriterService.findCouponInWriter(id));
```

Reply:
우와 멋져요 조조
Optional은 잘 사용하기 정말 어려운 것 같아요 😂